### ISSUE TITLE
[v2 QUESTION]: SpectralFNN functionality 

### ISSUE THREAD

**Issue by Rhys-McAlister** [chemprop/chemprop#829]
How does the SpectralFNN predictor model function? I'm not sure how to pass the correct dimensions to this module:

```
UserWarning: Using a target size (torch.Size([64, 1801])) that is different to the input size (torch.Size([64, 1])). This will likely lead to incorrect results due to broadcasting. Please ensure they have the same size.
  return F.mse_loss(preds, targets, reduction="none")
```

I can see that there is an n_targets parameter but I can't access this without just changing the file

**Comment by am2145** [comment]
We don't have full support for spectral predictions at the moment, but you can access the `n_tasks` parameter when initializing the FFN which should get the dimensions correct. For example: `ffn = nn.SpectralFFN(n_tasks=1801)`. To match the v1 workflow, the target (input) and predicted spectra are both expected to be sum-normalized. The SpectralFFN predictor handles the latter normalization. Normalizing the input spectra is not done by default at the moment so you'd have to perform this manually. We do plan to handle this for the user in future updates, but if you have the spectra absorbances in a dataframe, `df_input[target_columns] = df_input[target_columns].div(df_input[target_columns].sum(axis=1), axis=0)`  should do the job. 

I hope this helps, and let me know if you encounter any further issues.

**Comment by Rhys-McAlister** [comment]
Hello, I've followed your steps but am still getting a nan training loss immediately, is there any information I can provide to help troubleshoot this?

**Comment by am2145** [comment]
Hi Rhys, 

When you create the train, validation, and test datasets, can you check if a scaler is being applied? It would look like `scaler = train_dset.normalize_targets()`. Since you are normalizing the spectra, I would disable further scaling on the dataset that is applied by default for tasks like regression and see if that helps. 

Additionally, we could double check the metric to see if it's a potential issue there. From your first post, I'm assuming it's MSE. Is this correct? 

If you still are encountering NaN training losses after this, then it would be helpful to have a look at the input data if you are able to share a small example.

**Comment by Rhys-McAlister** [comment]
Hi, after removing the scalers I was still getting NaNs and so I just added a small constant (1e-3)ish to every row and that seems to fix the NaN issue for now

**Comment by am2145** [comment]
Glad that it's working now. For our information going forward with the full spectral implementation in v2, were there any negative or zero values in the input data? v1 filtered these out similarly to what you did here, so that may be what was causing the issue.

**Comment by JacksonBurns** [comment]
Closing for lack of user in put - please re-open or open a new issues if something arises.
